versioned/transfer/src/main/java/org/projectnessie/versioned/transfer/ExportDatabaseAdapter.java

```diff
@@ -97,17 +100,29 @@ private HeadsAndForkPoints scanAllReferences(Consumer<CommitLogEntry> commitHand
           .map(ReferenceInfo::getHash)
           .forEach(
               head -> {
-                try (Stream<CommitLogEntry> commits = databaseAdapter.commitLog(head)) {
-                  for (Iterator<CommitLogEntry> commitIter = commits.iterator();
-                      commitIter.hasNext(); ) {
-                    CommitLogEntry commit = commitIter.next();
-                    if (!identify.handleCommit(commit)) {
-                      break;
+                Deque<Hash> commitsToProcess = new ArrayDeque<>();
+                commitsToProcess.offerFirst(head);
```

Comment:
`Queue.add(head)` would be more intuitive (same on line 118), WDYT?

"offer" methods return a value, which we do not check, so it feels awkward in this case (even though they should not fail in practice).